### PR TITLE
Create url entry text box

### PR DESCRIPTION
**Pull request by harry48225**
I've added the url entry text box to match the design.

When clicked the place holder text fades, and then when focus is lost it reappears as long as nothing has been typed.

image

### CODE REVIEW

File: app/src/components/urlEntry.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      transition: visibility 0s linear 0s, opacity 300ms;
+    }
+
+    &.fadeOut {
+      visibility: hidden;
+      opacity: 0;
+      transition: visibility 0s linear 300ms, opacity 300ms;
+    }
+  }
+`;
```
Comment (BWBellairs) [inline app/src/components/urlEntry.tsx]: Can this be moved into it's own scss file please?

**Review by BWBellairs — COMMENTED** [review]
The animation looks nice but could we switch the textbox to use the `placeholder` attribute please?

File: app/src/components/urlEntry.tsx (hunk 1 above)
Comment (harry48225) [reply app/src/components/urlEntry.tsx]: It's actually better practice to have it in the js like this.

Styled components is a css-in-js library. Keeping the styles and the components together is actually good practice and it also allows you to use JavaScript variables in the styling. Also it means styles a scoped to their components so you don't get leaky CSS everywhere.

**Review by BWBellairs — APPROVED** [review]
Thank you, I like the changes.

**Pull request merged**